code/iaas/logic/src/main/java/io/cattle/platform/process/instance/InstanceAllocate.java

```diff
@@ -109,12 +111,30 @@ protected HandlerResult postEvent(ProcessState state, ProcessInstance process, M
         }
 
         List<Volume> volumes = getObjectManager().children(instance, Volume.class);
+        Map<String, Object> dataVolumeMounts = DataAccessor.field(instance, DATA_VOLUME_MOUNTS, Map.class);
+
+        if (dataVolumeMounts != null) {
+            for (Map.Entry<String, Object> entry : dataVolumeMounts.entrySet()) {
+                Volume v = getObjectManager().loadResource(Volume.class, ((Integer) entry.getValue()).longValue());
+                if (v != null) {
+                    volumes.add(v);
+                }
+            }
+        }
 
         for (Volume v : volumes) {
             allocate(v, state.getData());
         }
 
         volumes = getObjectManager().children(instance, Volume.class);
+        if (dataVolumeMounts != null) {
```

Comment:
This feels like deja vu, what is the difference from this block and the one above?